### PR TITLE
Fix mono alc from gchandle

### LINKED ISSUE #62052: [mono][bug] mono_alc_from_gchandle return wrong value?
### Description

I am embadding monovm,  I write a function `mono_alc_load_assembly_with_partial_name` extend from `mono_assembly_load_with_partial_name` with `alc` argument

when call `mono_image_open_from_data_alc` to load `MonoImage` the `mono_alc_from_gchandle` get the wrong `MonoAssemblyLoadContext` value

### Reproduction Steps

```c#
// I implemented AssemblyLoadContext:
public class PluginLoadContext : AssemblyLoadContext
  {
      
      public PluginLoadContext(string? name, bool isCollectible) : base(name, isCollectible)
      {
      }

      protected override Assembly? Load(AssemblyName assemblyName)
      {
          return PluginLoader.LoadAssembly(assemblyName.Name);
      }
  }
```

```c
// like mono_assembly_load_with_partial_name just pass our ALC, not use default alc
MonoAssembly* 
mono_alc_load_assembly_with_partial_name (void* native_alc, const char *name, MonoImageOpenStatus *status)
{
	MonoAssembly *result;
	MONO_ENTER_GC_UNSAFE;
	MonoImageOpenStatus def_status;
	if (!status)
		status = &def_status;
	MonoAssemblyLoadContext* alc = (MonoAssemblyLoadContext*)native_alc;
	if (!alc)
		alc = mono_alc_get_default();
	result = mono_assembly_load_with_partial_name_internal (name, alc, status);
	MONO_EXIT_GC_UNSAFE;
	return result;
}

// get MonoAssemblyLoadContext though `_nativeAssemblyLoadContext` field
// the internal call
MonoObject* ScriptingManager::LoadAssembly(MonoObject* alc, MonoString* assemblyName)
{
	void* NativeALC = GetNativeALC(alc);
	if (!NativeALC)
		return nullptr;
	MonoImageOpenStatus LoadStatus = MonoImageOpenStatus::MONO_IMAGE_OK;
	const char* AssemblyName = mono_string_to_utf8(assemblyName);
	MonoAssembly* Assembly = mono_alc_load_assembly_with_partial_name(NativeALC, AssemblyName, &LoadStatus);
	mono_free((void*)AssemblyName);
	return (MonoObject*)mono_assembly_get_object(mono_domain_get(), Assembly);
}
```
our code call step
``` c#
// in c#

PluginLoaderContent alc = new PluginLoaderContent ("Engine", true);
// call internal call ScriptingManager::LoadAssembly 
LoadAssembly(alc, "UnrealEngine");

```

image

the native stack variable

1. new PluginLoaderContent  create native ALC, and the alc native address is `0x000001c4bfdf51a0`

image

2. in the `ScriptingManager::LoadAssembly`, we can see the MonoObject* alc and native alc value is correct.
image

3. but in the mono_alc_from_gchandle the value is wong: 
image

4. the full call statck:
image

### Expected behavior

work correctly

### Actual behavior

crash

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

build the latest version of release/6.0 branch 
x64 windows desktop

### Other information

no

### PR DESCRIPTION
**Pull request by srxqds**
fix: https://github.com/dotnet/runtime/issues/62052

### CODE REVIEW

File: src/mono/mono/metadata/assembly-load-context.c [hunk 1]
```diff
@@ -424,11 +424,19 @@ mono_alc_from_gchandle (MonoGCHandle alc_gchandle)
...
+		g_assert (alc_class);
+		resolve = mono_class_get_field_from_name (alc_class, "_nativeAssemblyLoadContext");
+
+	MONO_STATIC_POINTER_INIT_END (MonoClassField, resolve)
+
+	g_assert (resolve);
+
+	MonoAssemblyLoadContext *alc = NULL;
+	mono_field_get_value(mono_gchandle_get_target_internal(alc_gchandle), resolve, &alc);
+	return alc;
```
Comment (vargaz) [inline src/mono/mono/metadata/assembly-load-context.c:439]: Formatting.

**Review by vargaz — APPROVED** [review]

File: src/mono/mono/metadata/assembly-load-context.c [hunk 2]
```diff
@@ -424,11 +424,19 @@ mono_alc_from_gchandle (MonoGCHandle alc_gchandle)
...
-	MonoManagedAssemblyLoadContextHandle managed_alc = MONO_HANDLE_CAST (MonoManagedAssemblyLoadContext, mono_gchandle_get_target_handle (alc_gchandle));
-	g_assert (!MONO_HANDLE_IS_NULL (managed_alc));
-	MonoAssemblyLoadContext *alc = MONO_HANDLE_GETVAL (managed_alc, native_assembly_load_context);
-	HANDLE_FUNCTION_RETURN_VAL (alc);
+	MONO_STATIC_POINTER_INIT (MonoClassField, resolve)
+	
+		MonoClass *alc_class = mono_class_get_assembly_load_context_class ();
+		g_assert (alc_class);
+		resolve = mono_class_get_field_from_name (alc_class, "_nativeAssemblyLoadContext");
+
```
Comment (vargaz) [inline src/mono/mono/metadata/assembly-load-context.c:432]: Needs to call mono_class_get_field_from_name_full ().

File: src/mono/mono/metadata/assembly-load-context.c (hunk 1 above)
Comment (vargaz) [reply src/mono/mono/metadata/assembly-load-context.c:439]: Needs to call mono_field_get_value_internal () instead.

**Comment by lateralusX** [comment]
Compile error:

```
 /__w/1/s/src/mono/mono/metadata/assembly-load-context.c:431:27: error: too few arguments to function ‘mono_class_get_field_from_name_full’
    431 |                 resolve = mono_class_get_field_from_name_full (alc_class, "_nativeAssemblyLoadContext");
        |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  In file included from /__w/1/s/src/mono/mono/mini/../../mono/metadata/handle.h:26,
                   from /__w/1/s/src/mono/mono/mini/../../mono/metadata/exception-internals.h:11,
                   from /__w/1/s/src/mono/mono/metadata/assembly-load-context.c:7:
  /__w/1/s/src/mono/mono/mini/../../mono/metadata/class-internals.h:1227:1: note: declared here
   1227 | mono_class_get_field_from_name_full (MonoClass *klass, const char *name, MonoType *type);
        | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

**Review by lateralusX — APPROVED** [review]

**Pull request merged**